src/conmon.c

```diff
@@ -664,29 +680,17 @@ static gboolean conn_sock_cb(int fd, GIOCondition condition, gpointer user_data)
 	ssize_t num_read = 0;
 
 	if ((condition & G_IO_IN) != 0) {
-		num_read = splice(fd, NULL, masterfd_stdin, NULL, 1 << 20, 0);
-		if (num_read > 0)
+		char buf[CONN_SOCK_BUF_SIZE];
```

Comment:
@giuseppe I saw the call to splice hang long enough to have podman timeout. It's probably an i/o scheduler problem but this fixes it. I tried using NONBLOCK for the splice and that didn't work either. Can you think of anything that would prevent the hang under i/o pressure?

Reply:
unfortunately this can hang indefinitely.

If the data written to `fd` is not read, then conmon will hang on the `write_all`.

I've used the splice to address: https://github.com/containers/conmon/issues/99.

Would lower the `len` argument to splice fix the issue you've seen?  It is currently set to 1M but I think it is fine if we get down to 4096 bytes

Reply:
okay I'll try this, thanks!